src/EntityAutocompleteMatcher.php

```diff
@@ -27,6 +28,12 @@ public function getMatches($target_type, $selection_handler, $selection_settings
 
     $handler = $this->selectionManager->getInstance($options);
 
+    // If the handler cannot be found, returns the default one.
+    if ($handler instanceof Broken) {
```

Comment:
@vincent-gao 
Just wondering under what circumstances is Drupal not able to load the actual EntityReferenceSelection plugin? Are we covering up some other config or code problem? Which plugin was it trying to load when it returned "Broken" instead?

Reply:
I read the bug report ticket but couldn't see any clues.

Reply:
@vincent-gao 
I think I see why it cannot load the EntityReferenceSelection handler. There's no hander called "tide_core". 

It should be looking for "tide:node" not "tide_core" at line 20.

The handler is defined here:
https://github.com/dpc-sdp/tide_core/blob/develop/src/Plugin/EntityReferenceSelection/TideNodeSelection.php 

```php
/**
 * Provides labels for the node entity type.
 *
 * @EntityReferenceSelection(
 *   id = "tide:node",
 *   label = @Translation("Tide node selection"),
 *   entity_types = {"node"},
 *   group = "tide",
 *   weight = 1,
 *   base_plugin_label = "Tide"
 * )
 */
```

If you fix that, it should no longer return the "broken" handler as a fallback I think. Can you try that out?

Reply:
Thanks @christopher-hopper ! I think it works.